Fix Jinja conditional

priteau opened the pull request:
## Description

A change to the bool filter [1] is impacting generated configuration. Before ansible-core 2.19, `chrony_local_stratum | bool` would evaluate to true with the default value of `chrony_local_stratum: 10`. Now it evaluates to false:

    [DEPRECATION WARNING]: The `bool` filter coerced invalid value 10
    (int) to False. This feature will be removed from ansible-core
    version 2.23.

[1] https://github.com/ansible-community/ansible-build-data/blob/main/12/CHANGELOG-v12.md#ansible-core-9

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

mrlesmithjr reviewed (changes requested):
Thanks for addressing the ansible-core 2.19 `bool` filter deprecation - this is an important fix!

A few items to address before merging:

## CI Failure
The build is failing on ubuntu2204/Python 3.9. Please investigate and resolve before we can merge.

## Design Consideration

The current fix works but conflates "enable feature" with "stratum value > 0". The original code was using `bool` incorrectly (relying on integer truthiness), which worked by accident.

Two options to consider:

**Option A (Preferred):** Add a dedicated enable flag for semantic clarity:
```yaml
# defaults/main.yml
chrony_enable_local_stratum: true
chrony_local_stratum: 10
```
```jinja
{% if chrony_enable_local_stratum | default(false) %}
local stratum {{ chrony_local_stratum }}
{% endif %}
```

**Option B:** Keep your approach but document the behavior in `defaults/main.yml`:
```yaml
# Set to a value > 0 to enable local stratum mode
# Set to 0 or leave undefined to disable
chrony_local_stratum: 10
```

Either approach is acceptable - just want to ensure the behavior is clear for users.

## Documentation
Please add a brief note about this change for users who may have relied on the previous (accidental) behavior.

Thanks again for the contribution!

priteau committed:
Fix Jinja conditional

A change to the bool filter [1] is impacting generated configuration.
Before ansible-core 2.19, `chrony_local_stratum | bool` would evaluate
to true with the default value of `chrony_local_stratum: 10`. Now it
evaluates to false:

    [DEPRECATION WARNING]: The `bool` filter coerced invalid value 10
    (int) to False. This feature will be removed from ansible-core
    version 2.23.

[1] https://github.com/ansible-community/ansible-build-data/blob/main/12/CHANGELOG-v12.md#ansible-core-9

mrlesmithjr reviewed (approved):
Thanks for the updates! The PR now looks good:

- ✅ CI passing
- ✅ Documentation updated in `defaults/main.yml` clarifying `null` or `0` disables the feature
- ✅ Thorough commit message with ansible-core changelog reference

LGTM - ready to merge!
